bin/cylc-cat-log

```diff
@@ -407,31 +415,29 @@ def main():
             if batchview_cmd:
                 cmd.append('--remote-arg=%s' % quote(batchview_cmd))
             cmd.append(suite_name)
-            capture = (mode == 'edit')
+            is_edit_mode = (mode == 'edit')
             try:
                 proc = remote_cylc_cmd(
-                    cmd, user, host, capture_process=capture,
+                    cmd, user, host, capture_process=is_edit_mode,
                     manage=(mode == 'tail'))
             except KeyboardInterrupt:
                 # Ctrl-C while tailing.
                 pass
             else:
-                if capture:
+                if is_edit_mode:
                     # Write remote stdout to a temp file for viewing in editor.
                     # Only BUFSIZE bytes at a time in case huge stdout volume.
-                    out = mkstemp(dir=glbl_cfg().get_tmpdir())[1]
-                    with open(out, 'wb') as outf:
+                    out = NamedTemporaryFile()
+                    data = proc.stdout.read(BUFSIZE)
+                    while data:
+                        out.write(data)
                         data = proc.stdout.read(BUFSIZE)
-                        while data:
-                            outf.write(data)
-                            data = proc.stdout.read(BUFSIZE)
-                    os.chmod(out, S_IRUSR)
+                    os.chmod(out.name, S_IRUSR)
+                    out.seek(0, 0)
         else:
             # Local task job or local job log.
-            logpath = os.path.normpath(os.path.join(
-                glbl_cfg().get_derived_host_item(
-                    suite_name, "suite job log directory"),
-                point, task, options.submit_num, options.filename))
+            logpath = os.path.normpath(get_suite_run_job_dir(
+                suite_name, point, task, options.submit_num, options.filename))
```

Comment:
This code style is inconsistent with lines 396-404 above.  I prefer the compact style.

Reply:
Changed above to match the compact style here. 3 lines instead of 2 though.